homeassistant/components/solarlog/config_flow.py

```diff
@@ -0,0 +1,85 @@
+"""Config flow for solarlog integration."""
+from requests.exceptions import HTTPError, Timeout
+from sunwatcher.solarlog.solarlog import SolarLog
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.util import slugify
+
+from .const import DEFAULT_HOST, DEFAULT_NAME, DOMAIN
+
+
+@callback
+def solarlog_entries(hass: HomeAssistant):
+    """Return the hosts already configured."""
+    return set(
+        entry.data[CONF_HOST] for entry in hass.config_entries.async_entries(DOMAIN)
+    )
+
+
+class SolarLogConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for solarlog."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._errors = {}
+
+    def _host_in_configuration_exists(self, user_input) -> bool:
+        """Return True if host exists in configuration."""
+        host = user_input.get(CONF_HOST, DEFAULT_HOST)
+        if host in solarlog_entries(self.hass):
+            return True
+        return False
+
+    async def _test_connection(self, user_input=None):
+        """Check if we can connect to the Solar-Log device."""
+        host = user_input.get(CONF_HOST, DEFAULT_HOST)
+        try:
+            SolarLog(f"http://{host}")
```

Comment:
This is doing I/O and so cannot be run inside an async function. Run it in the executor instead:

```
await self.hass.async_add_executor_job(SolarLog, f"http://{host}")
```

Reply:
Fixed